extensions/WSLExtension/Services/WslManager.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Serilog;
+using Windows.System.Threading;
+using WSLExtension.ClassExtensions;
+using WSLExtension.Contracts;
+using WSLExtension.DistributionDefinitions;
+using WSLExtension.Helpers;
+using WSLExtension.Models;
+using static WSLExtension.Constants;
+
+namespace WSLExtension.Services;
+
+public class WslManager : IWslManager
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(WslManager));
+
+    private readonly PackageHelper _packageHelper = new();
+
+    private readonly TimeSpan _oneMinutePollingInterval = TimeSpan.FromMinutes(1);
+
+    private readonly WslRegisteredDistributionFactory _wslRegisteredDistributionFactory;
+
+    private readonly IWslServicesMediator _wslServicesMediator;
+
+    private readonly IDistributionDefinitionHelper _definitionHelper;
+
+    private readonly List<WslComputeSystem> _registeredWslDistributions = new();
+
+    public event EventHandler<HashSet<string>>? DistributionStateSyncEventHandler;
+
+    private Dictionary<string, DistributionDefinition>? _distributionDefinitionsMap;
+
+    private ThreadPoolTimer? _timerForUpdatingDistributionStates;
+
+    public WslManager(
+        IWslServicesMediator wslServicesMediator,
+        WslRegisteredDistributionFactory wslDistributionFactory,
+        IDistributionDefinitionHelper distributionDefinitionHelper)
+    {
+        _wslRegisteredDistributionFactory = wslDistributionFactory;
+        _wslServicesMediator = wslServicesMediator;
+        _definitionHelper = distributionDefinitionHelper;
+        StartDistributionStatePolling();
+    }
+
+    /// <inheritdoc cref="IWslManager.GetAllRegisteredDistributionsAsync"/>
+    public async Task<List<WslComputeSystem>> GetAllRegisteredDistributionsAsync()
+    {
+        // The list of compute systems in Dev Home is being refreshed, so remove any old
+        // subscriptions
+        for (var i = _registeredWslDistributions.Count - 1; i >= 0; i--)

```

Comment:
Perf: Call _registeredWslDistributions.Clear after the loop to remove all elements instead of removing one by one. And then the loop can be "foreach" or _registeredWslDistributions.ForEach(distribution => )distribution.RemoveSubscriptions()).

Reply:
sure will update